sdk/storage/storage-blob/review/generated/src/models/index.d.ts

```diff
@@ -0,0 +1,6713 @@
+}
+/**
+ * Optional Parameters.
+ */
+export interface BlobSetHTTPHeadersOptionalParams extends coreHttp.RequestOptionsBase {
+    /**
+     * The timeout parameter is expressed in seconds. For more information, see <a
+     * href="https://docs.microsoft.com/en-us/rest/api/storageservices/fileservices/setting-timeouts-for-blob-service-operations">Setting
+     * Timeouts for Blob Service Operations.</a>
+     */
+    timeoutParameter?: number;
+    /**
+     * Provides a client-generated, opaque value with a 1 KB character limit that is recorded in the
+     * analytics logs when storage analytics logging is enabled.
+     */
+    requestId?: string;
+    /**
+     * Additional parameters for the operation
+     */
+    blobHTTPHeaders?: BlobHTTPHeaders;
+    /**
+     * Additional parameters for the operation
+     */
+    leaseAccessConditions?: LeaseAccessConditions;
+    /**
+     * Additional parameters for the operation
+     */
+    modifiedAccessConditions?: ModifiedAccessConditions;
+}
+/**
+ * Optional Parameters.
+ */
+export interface BlobSetMetadataOptionalParams extends coreHttp.RequestOptionsBase {
+    /**
+     * The timeout parameter is expressed in seconds. For more information, see <a
+     * href="https://docs.microsoft.com/en-us/rest/api/storageservices/fileservices/setting-timeouts-for-blob-service-operations">Setting
+     * Timeouts for Blob Service Operations.</a>
+     */
+    timeoutParameter?: number;
+    /**
+     * Optional. Specifies a user-defined name-value pair associated with the blob. If no name-value
+     * pairs are specified, the operation will copy the metadata from the source blob or file to the
+     * destination blob. If one or more name-value pairs are specified, the destination blob is
+     * created with the specified metadata, and metadata is not copied from the source blob or file.
+     * Note that beginning with version 2009-09-19, metadata names must adhere to the naming rules
+     * for C# identifiers. See Naming and Referencing Containers, Blobs, and Metadata for more
+     * information.
+     */
+    metadata?: {
+        [propertyName: string]: string;
+    };
+    /**
+     * Provides a client-generated, opaque value with a 1 KB character limit that is recorded in the
+     * analytics logs when storage analytics logging is enabled.
+     */
+    requestId?: string;
+    /**
+     * Additional parameters for the operation
+     */
+    leaseAccessConditions?: LeaseAccessConditions;
+    /**
+     * Additional parameters for the operation
+     */
+    cpkInfo?: CpkInfo;
+    /**
+     * Additional parameters for the operation
+     */
+    modifiedAccessConditions?: ModifiedAccessConditions;
+}
+/**
+ * Optional Parameters.
+ */
+export interface BlobAcquireLeaseOptionalParams extends coreHttp.RequestOptionsBase {
+    /**
+     * The timeout parameter is expressed in seconds. For more information, see <a
+     * href="https://docs.microsoft.com/en-us/rest/api/storageservices/
```

Comment:
in the other file in this PR we use a string enum and here we use a union of string literals. Should we be consistent?